tests/jax/components/building/datasets_test.py

```diff
@@ -0,0 +1,201 @@
+        self.simple_server = reverb.Server(
+            [
+                reverb.Table.queue(
+                    name="table_0",
+                    max_size=100,
+                    signature=adder_signature_fn(env_spec, {}),
+                )
+            ]
+        )
+        data_server_client = SimpleNamespace(
+            server_address=f"localhost:{self.simple_server.port}"
+        )
+        trainer_id = "table_0"
+        self.store = SimpleNamespace(
+            data_server_client=data_server_client, trainer_id=trainer_id
+        )
+
+
+@pytest.fixture
+def mock_builder() -> MockBuilder:
+    """Create builder mock"""
+    return MockBuilder()
+
+
+@pytest.fixture
+def transition_dataset() -> TransitionDataset:
+    config = TransitionDatasetConfig()
+    config.sample_batch_size = 512
+    config.prefetch_size = None
+    config.num_parallel_calls = 24
+    config.max_in_flight_samples_per_worker = None
+    config.postprocess = None
+
+    transition_dataset = TransitionDataset(config=config)
+    return transition_dataset
+
+
+@pytest.fixture
+def trajectory_dataset() -> TrajectoryDataset:
+    config = TrajectoryDatasetConfig()
+    config.sample_batch_size = 512
+    config.max_in_flight_samples_per_worker = 1024
+    config.num_workers_per_iterator = -2
+    config.max_samples_per_stream = -2
+    config.rate_limiter_timeout_ms = -2
+    config.get_signature_timeout_secs = None
+
+    trajectory_dataset = TrajectoryDataset(config=config)
+    return trajectory_dataset
+
+
+def test_init_transition_dataset(transition_dataset: TransitionDataset) -> None:
+    """Test initiator of TransitionDataset component"""
+    assert transition_dataset.config.sample_batch_size == 512
+    assert transition_dataset.config.prefetch_size is None
+    assert transition_dataset.config.num_parallel_calls == 24
+    assert transition_dataset.config.max_in_flight_samples_per_worker is None
+    assert transition_dataset.config.postprocess is None
+
+
+def test_on_building_trainer_dataset_transition_dataset(
+    mock_builder: MockBuilder,
+) -> None:
+    """Test on_building_trainer_dataset of TransitionDataset Component
+
+    Args:
+        mock_builder: Builder
+    """
+    transition_dataset = TransitionDataset()
+    transition_dataset.on_building_trainer_dataset(builder=mock_builder)
+
+    # mock_builder.store.dataset_iterator._dataset._map_func._func(1)._dataset \
+    # is needed to check the parameters i.e. obtain the dataset from the \
+    # tf.data.Dataset dataset iterator
+
+    dataset = mock_builder.store.dataset_iterator._dataset._map_func._func(1)._dataset
+    assert (
+        dataset._input_dataset._server_address
+        == mock_builder.store.data_server_client.server_address
+    )
+    assert dataset._input_dataset._table == mock_builder.store.trainer_id
+    assert dataset._batch_size == transition_dataset.config.sample_batch_size
+    assert (
+        dataset._input_dataset._max_in_flight_samples_per_worker
```

Comment:
I don't think we can always assume `_max_in_flight_samples_per_worker = 2 * transition_dataset.config.sample_batch_size`.

Reply:
Yeah, you were right it is the case where max_in_flight_sample_per_worker=None 
```
if max_in_flight_samples_per_worker is None and batch_size is None:
    max_in_flight_samples_per_worker = 100
  elif max_in_flight_samples_per_worker is None:
    max_in_flight_samples_per_worker = 2 * batch_size
```
I will make 3 cases for it (batch_size and non max_in.., non batch_size and non max_in, max_in_flight non none)

Reply:
Done!